presto-orc/src/main/java/com/facebook/presto/orc/reader/DecimalStreamReader.java

```diff
@@ -183,47 +125,60 @@ private void readStreamsData(DecimalType decimalType)
                 }
 
                 if (decimalType.isShort()) {
-                    decimalStream.nextLongVector(nextBatchSize, shortDecimalVector, nullVector);
+                    decimalStream.nextShortDecimalVector(nextBatchSize, builder, decimalType, scaleVector, nullVector);
                 }
                 else {
-                    decimalStream.nextBigIntegerVector(nextBatchSize, longDecimalVector, nullVector);
+                    decimalStream.nextLongDecimalVector(nextBatchSize, builder, decimalType, scaleVector, nullVector);
                 }
 
                 scaleStream.nextLongVector(nextBatchSize, scaleVector, nullVector);
             }
         }
+
+        readOffset = 0;
+        nextBatchSize = 0;
+
+        return builder.build();
     }
 
-    private Block buildDecimalsBlock(DecimalType decimalType)
-            throws OrcCorruptionException
+    private void openRowGroup()
```

Comment:
It would be a lot easier to read this diff, if the reordering of the methods were in a separate commit.  Also, does this order now follow the same order as the other ORC stream readers?